sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/serializer/AvroSerializer.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.experimental.serializer;
+
+import com.azure.core.util.serializer.ObjectSerializer;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Mono;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * Generic interface covering basic Avro serialization and deserialization methods.
+ */
+public interface AvroSerializer extends ObjectSerializer {
```

Comment:
Made `AvroSerializer` a sub-interface to `ObjectSerializer` so it can extend at will in the future without cause breaking changes.